learning_table_tennis_from_scratch/ppo_config.py

```diff
@@ -1,52 +1,77 @@
-import sys
-import os
 import json
+import os
+import sys
+import typing
+
+
+ConfigDict = typing.Dict[str, typing.Any]
 
 
 class PPOConfig:
+    """Configuration for stable_baselines3 PPO."""
 
-    __slots__ = (
+    # These parameters are directly passed to stable_baselines3.PPO, see the
+    # documentation there for their meaning.
+    _ppo_params = (
         "gamma",
         "n_steps",
         "ent_coef",
         "learning_rate",
-        "cliprange",
-        "cliprange_vf",
+        "clip_range",
+        "clip_range_vf",
         "vf_coef",
         "max_grad_norm",
-        "lam",
-        "nminibatches",
-        "noptepochs",
+        "gae_lambda",
+        "batch_size",
+        "n_epochs",
+    )
+
+    # Additional parameters
```

Comment:
what is the difference between ppo params and additional params ?

Reply:
`_ppo_params` are arguments of `stable_baselines3.PPO` that are directly passed to the constructor when instantiating the PPO class.  The "additional parameters" have to be passed in different ways (e.g. when configuring the logger or when calling the `learn()` method).  The reason for separating them is that it allows to get a dictionary with only the `_ppo_param`-part (see `get_ppo_params()`) so it can easily be passed to `PPO()` as kwargs.